File: src/main/java/org/carlosalcina/java/rpg/Personaje.java
package org.carlosalcina.java.rpg;

public class Personaje {
    protected String nombre;
    protected int puntosVida;

    public Personaje(String nombre, int puntosVida) {
        this.nombre = nombre;
        this.puntosVida = puntosVida;
    }

    public String getNombre() { return nombre; }
    public int getPuntosVida() { return puntosVida; }

    public void recibirDanio(int danio) {
        puntosVida -= danio;
        if (puntosVida < 0) puntosVida = 0;
    }
}

File: src/main/java/org/carlosalcina/kotlin/rpg/Personaje.kt
package org.carlosalcina.kotlin.rpg

open class Personaje(val nombre: String, var puntosVida: Int) {
    fun recibirDanio(danio: Int) {
        puntosVida -= danio
        if (puntosVida < 0) puntosVida = 0
    }
}